tests/acceptance/bootstrap/SharingNgContext.php

```diff
@@ -2091,4 +2094,29 @@ public function theJsonResponseShouldContainTheFollowingShares(TableNode $table)
 			);
 		}
 	}
+
+	/**
+	 * @When /^user "([^"]*)" gets federated permissions list for (folder|file) "([^"]*)" of the space "([^"]*)" using the Graph API$/
+	 *
+	 * @param string $user
+	 * @param string $fileOrFolder (file|folder)
+	 * @param string $resource
+	 * @param string $space
+	 *
+	 * @return void
+	 * @throws Exception
+	 */
+	public function userGetsFederatedPermissionsListForFileOfTheSpaceUsingTheGraphApi(
+		string $user,
+		string $fileOrFolder,
+		string $resource,
+		string $space
+	): void {
+		$query = '?$filter=@libre.graph.permissions.roles.allowedValues'
+			. '/rolePermissions/any(p:contains(p/condition,+\'@Subject.UserType=="Federated"\'))'
+			. '&$select=@libre.graph.permissions.roles.allowedValues';
```

Comment:
```suggestion
		$query = [
			'$filter' => '@libre.graph.permissions.roles.allowedValues/rolePermissions/any(p:contains(p/condition,+\'@Subject.UserType=="Federated"\'))',
			'$select' => '@libre.graph.permissions.roles.allowedValues'
		]
```

Reply:
give problem `p/condition,+\'@Subject` **+** should not be encoded to get proper reply